Lex and Parse refactor

corwin opened the pull request.

Commit:
WIP. Initial skeleton of if/notif lexer and parsing.

Commit:
WIP. Initial skeleton of if/notif lexer and parsing.

JessicaSachs commented:
Super WIP at this point... still re-implementing each, for, component, and the custom options Easybars can be configured with.

corwin committed:
Fix handling of spurious {{/if}}

Upon encountering a non-matching end tag, just ignore it and keep going.

corwin committed:
Fix handling of spurious {{/if}}

Upon encountering a non-matching end tag, just ignore it and keep going.

Commit:
Merge pull request #8 from corwin/patch-1

This fixes the spec related to dropping bogus closing tags.

Commit:
Tokens as objects instead of arrays.

Commit:
Tokens as objects instead of arrays.

Commit:
JS: makeToken should allow for anonymous options that are token specific to be passed in

Commit:
JS: makeToken should allow for anonymous options that are token specific to be passed in

corwin committed:
Cleaned up handling of negated if.

corwin committed:
Cleaned up handling of negated if.

Commit:
JS: Deal with an actions value in terms of the n arguments it could have

Commit:
JS: Deal with an actions value in terms of the n arguments it could have

Commit:
JS: Put negated check inside of if handler

Commit:
JS: Put negated check inside of if handler

Commit:
JS: Scoping negation handling to if blocks only

Commit:
JS: Scoping negation handling to if blocks only

Commit:
JS: Scoping negation handling to if blocks only

Commit:
JS: Scoping negation handling to if blocks only

Commit:
Merge branch 'jess/corwin/lex_and_parse_refactor' of github.com:seebigs/easybars into jess/corwin/lex_and_parse_refactor

corwin committed:
corwin: rewrote lexer

  index.js:
    Rewrote the lexer to generate its own token stream in preparation
    for supporting re-interpolation.

  test/specs/edgecase.spec.js:
    Added test of a template string with nothing to interpolate.

corwin committed:
corwin: rewrote lexer

  index.js:
    Rewrote the lexer to generate its own token stream in preparation
    for supporting re-interpolation.

  test/specs/edgecase.spec.js:
    Added test of a template string with nothing to interpolate.

corwin committed:
corwin: implemented multiple interpolation

  index.js:
  Added a parse() method which takes a string and data object, this is
    a wrapper for parseTokens().
  Implemented multiple interpolation by calling parse() on the result
    an interpolation lookup.
  Rewrote parseTokens() to have (hopefully) clearer structure as well
    as internal dispatching.

  test/specs/sections/combination.spec.js:
  Enabled at the top level.
  Added test of multiple interpolation.
  Enabled nested-if test.

corwin committed:
corwin: implemented multiple interpolation

  index.js:
  Added a parse() method which takes a string and data object, this is
    a wrapper for parseTokens().
  Implemented multiple interpolation by calling parse() on the result
    an interpolation lookup.
  Rewrote parseTokens() to have (hopefully) clearer structure as well
    as internal dispatching.

  test/specs/sections/combination.spec.js:
  Enabled at the top level.
  Added test of multiple interpolation.
  Enabled nested-if test.

Commit:
JS: Cleanup indentation issues.. fix small typos

Commit:
JS: Cleanup indentation issues.. fix small typos

Commit:
JS: Remove ES6 const. Prefer ES5.

Commit:
JS: Remove ES6 const. Prefer ES5.

Commit:
JS: Move inner-functions to the top of the function scope. Fix indentation.

Commit:
JS: Move inner-functions to the top of the function scope. Fix indentation.

Commit:
JS: Cleaning up apply calls. Prefer empty object for this context instead of null.

Commit:
JS: Cleaning up apply calls. Prefer empty object for this context instead of null.

Commit:
JS: Proper jsdoc style

Commit:
JS: Proper jsdoc style

Commit:
JS: For loop is almost done.

Commit:
JS: For loop is almost done.

corwin committed:
corwin: implemented for loop with nesting support

  index.js:
    Implemented for loop interpolation.

  for.spec.js:
    Enabled all but the encoding tests.

  combination.spec.js:
    Enabled nested for test.

corwin committed:
corwin: implemented for loop with nesting support

  index.js:
    Implemented for loop interpolation.

  for.spec.js:
    Enabled all but the encoding tests.

  combination.spec.js:
    Enabled nested for test.

corwin committed:
corwin: fixed bug in previous commit

  index.js:
    Fixed missed rename of findForTokens().
    Removed debug logging.

corwin committed:
corwin: fixed bug in previous commit

  index.js:
    Fixed missed rename of findForTokens().
    Removed debug logging.

corwin committed:
corwin: re-enabled tests
  test/specs/edgecase.spec.js:
    Re-enabled empty string test.

  test/specs/sections/if.spec.js:
    Re-enabled all recently disabled tests.

corwin committed:
corwin: re-enabled tests
  test/specs/edgecase.spec.js:
    Re-enabled empty string test.

  test/specs/sections/if.spec.js:
    Re-enabled all recently disabled tests.

corwin committed:
corwin: Cleanup and code motion prepartory to implementing encoding.
  index.js:
    Made parseTokens() an inner function of parse() so that when the
      encoding context is passed to parse(), it can be state rather
      than a parameter of parseTokens().
    Removed references token.data which ia a vestiage of the previous
      for implementation and which is never set.

corwin committed:
corwin: Cleanup and code motion prepartory to implementing encoding.
  index.js:
    Made parseTokens() an inner function of parse() so that when the
      encoding context is passed to parse(), it can be state rather
      than a parameter of parseTokens().
    Removed references token.data which ia a vestiage of the previous
      for implementation and which is never set.

corwin committed:
corwin: implemented encoding
  index.js:
    Implemented encoding by modifying the lexer to mark interpolate
      tokens which were {{{}}} delimtied, and modifying the parser to
      encode after interpolating if the interpolate token was marked.
    Added a call to escapeChars() before returningt he parsed string in
      parse().

  test/specs/sections/for.spec.js
  test/specs/sections/if.spec.js
    Enabled encoding tests.

corwin committed:
corwin: implemented encoding
  index.js:
    Implemented encoding by modifying the lexer to mark interpolate
      tokens which were {{{}}} delimtied, and modifying the parser to
      encode after interpolating if the interpolate token was marked.
    Added a call to escapeChars() before returningt he parsed string in
      parse().

  test/specs/sections/for.spec.js
  test/specs/sections/if.spec.js
    Enabled encoding tests.

corwin committed:
corwin: Implemented each loops.
  index.js:
    Implemented each by adding makeEach() to the lexer and map() to
      the parser.

  test/specs/sections/each.spec.js:
  test/specs/sections/combination.spec.js:
    Enabled each tests.

corwin committed:
corwin: Implemented each loops.
  index.js:
    Implemented each by adding makeEach() to the lexer and map() to
      the parser.

  test/specs/sections/each.spec.js:
  test/specs/sections/combination.spec.js:
    Enabled each tests.

corwin committed:
corwin: Fixed multiline matching bugs.
  index.js:
    Fixed token and action regexs to deal with newlines properly.

  test/specs/sections/if.spec.js:
    Added tests of prefixes and suffixes with newlines.

corwin committed:
corwin: Fixed multiline matching bugs.
  index.js:
    Fixed token and action regexs to deal with newlines properly.

  test/specs/sections/if.spec.js:
    Added tests of prefixes and suffixes with newlines.

corwin committed:
corwin: Enabled disabled edgecase test.

corwin committed:
corwin: Enabled disabled edgecase test.

corwin committed:
corwin: Fixed regex and interpolate bugs.
  index.js:
  Modified actionRE to properly handle white space at the start of
    an encode interpolation.
  Fixed bug in interpolate() when the value being interpolated is
    not a string and has a value which is considered false.

  test/specs/default.spec.js:
  Enabled some tests.

corwin committed:
corwin: Fixed regex and interpolate bugs.
  index.js:
  Modified actionRE to properly handle white space at the start of
    an encode interpolation.
  Fixed bug in interpolate() when the value being interpolated is
    not a string and has a value which is considered false.

  test/specs/default.spec.js:
  Enabled some tests.

Commit:
JS: Fix encoding, fix indentation, fix encoding inside of {{{{{vars}}}}}

Commit:
JS: Fix encoding, fix indentation, fix encoding inside of {{{{{vars}}}}}

Commit:
JS/CB: Easybars finishing touches. Bargaining to remove custom tags

Commit:
JS/CB: Easybars finishing touches. Bargaining to remove custom tags

Commit:
JS: Added conditional fixtures to perf test harness

Commit:
JS: Added conditional fixtures to perf test harness

Commit:
JS: killing eslint plugins that aren't actually needed

Commit:
JS: killing eslint plugins that aren't actually needed

corwin committed:
corwin: Fixed function doc formatting.

Commit:
JS: Move collapse into parse

Commit:
JS: Move collapse into parse

Commit:
JS: Remove support for sections in options.tags

Commit:
JS: Remove support for sections in options.tags

Commit:
JS: Remove unsupported options from the README

Commit:
JS: Remove unsupported options from the README

Commit:
JS: Collapsing new lines should not insert a space.

Commit:
Merge branch 'jess/corwin/lex_and_parse_refactor' of github.com:seebigs/easybars into jess/corwin/lex_and_parse_refactor

corwin committed:
Merge branch 'jess/corwin/lex_and_parse_refactor' of https://github.com/seebigs/easybars into jess/corwin/lex_and_parse_refactor

corwin committed:
Merge branch 'jess/corwin/lex_and_parse_refactor' of https://github.com/seebigs/easybars into jess/corwin/lex_and_parse_refactor

Commit:
JS: Tweaks to the regex to correctly handle new lines, tabs, etc

Commit:
JS: Remove weird non-ascii return character

Commit:
JS: {1,} is the same as + in regex

Commit:
JS: Additional regex fixes

Commit:
JS: Improving collapse spec. AAsserting collapse trims multiple trailing spaces
